primitives/state-machine/src/trie_backend_essence.rs

```diff
@@ -253,19 +254,13 @@ where
 		f: impl FnMut(Vec<u8>, Vec<u8>) -> bool,
 		allow_missing_nodes: bool,
 	) -> Result<bool> {
-		let mut child_root;
 		let root = if let Some(child_info) = child_info.as_ref() {
-			if let Some(fetched_child_root) = self.child_root(child_info)? {
-				child_root = H::Out::default();
-				// root is fetched from DB, not writable by runtime, so it's always valid.
-				child_root.as_mut().copy_from_slice(fetched_child_root.as_slice());
-
-				&child_root
-			} else {
-				return Ok(true)
+			match self.child_root(child_info)? {
+				Some(child_root) => child_root,
+				None => return Ok(false),
```

Comment:
```suggestion
				None => return Ok(true),
```

Reply:
(I did not check if returning false can be an issue, just returning true is aligned with the comment).

Reply:
It was `true` before, so you are right. My bad!